data_dictionary/data_dictionary.hh

```diff
@@ -55,11 +55,13 @@ class impl;
 class user_types_metadata;
 class keyspace_metadata;
 
+// Should not be returned directly to the user
```

Comment:
> I don't think we need such a comment [...] here.

Because I didn't understand at first that this exception is not meant to be seen by the "end user", I introduced a regression which I'm now fixing, so in order to mitigate this happening again I thought I'd be good to have some sort of a warning for whoever works with this exception. Of course now I understand all of it, but others may not. Perhaps I can reword this comment to make it more usable? Or maybe this message is redundant because everyone already knew it (except me)?

Reply:
But the thing is, this comment applies to every single except in the code **except** a few exceptions - the hierarchy of CQL exceptions that are just in CQL code, and the hierarchy of Alternator exceptions in Alternator. I don't think it would be practical to add this comment on all of them. Maybe if you think that this specific exception is particularly confusing we can indeed add a comment for it - but in that case I think it should be worded a bit less ambigously (since I think "user" has two different meanings, and also misses the fact that there are two different hierarchies of user-facing exceptions - CQL and Alternator).